pkg/modulewriter/modulewriter.go

```diff
@@ -106,6 +112,63 @@ func WriteDeployment(bp config.Blueprint, deploymentDir string) error {
 	return nil
 }
 
+func stageFiles(bp config.Blueprint, deplPath string) error {
+	staged := bp.StagedFiles()
+	if len(staged) == 0 {
+		return nil
+	}
+
+	// create staging directory
+	if err := os.MkdirAll(filepath.Join(deplPath, config.StagingDir), 0700); err != nil {
+		return err
+	}
+
+	errs := config.Errors{}
+	for src, dst := range staged {
+		// TODO: attribute error to the position in the blueprint
+		errs.Add(stageFile(deplPath, src, dst))
+	}
+	return errs.OrNil()
+}
+
+func doesExists(path string) (bool, error) {
+	_, err := os.Lstat(path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// stageFile copies a file from the src (absolute path) to the dst (relative to group directory in deployment directory)
+func stageFile(deplPath, absSrc, relDst string) error {
+	// relDst is relative to group folders ("../.ghpc/staged"),
+	// prepend any_group_dir to be "eaten" by ".."
+	absDst := filepath.Join(deplPath, "any_group_dir", relDst)
+	dstExists, err := doesExists(absDst)
```

Comment:
nit: can be combined with below if check.

Reply:
I don't see how.